package/MDAnalysis/analysis/msd.py

```diff
@@ -281,15 +281,34 @@ class EinsteinMSD(AnalysisBase):
         the MSD. Otherwise, use the simple "windowed" algorithm.
         The tidynamics package is required for `fft=True`.
         Defaults to ``True``.
+    non_linear : bool
+        If ``True``, calculates MSD for trajectory where frames are
+        non-linearly dumped. To use this set `fft=False`.
+        Defaults to ``False``.
+
+        .. versionadded:: 2.10.0
+
 
     Attributes
     ----------
     dim_fac : int
         Dimensionality :math:`d` of the MSD.
     results.timeseries : :class:`numpy.ndarray`
-        The averaged MSD over all the particles with respect to lag-time.
+        The averaged MSD over all the particles with respect to constant lag-time or
+        unique Δt intervals.
     results.msds_by_particle : :class:`numpy.ndarray`
-        The MSD of each individual particle with respect to lag-time.
+        The MSD of each individual particle with respect to constant lag-time or
+        unique Δt intervals.
+            for `non_linear=False`: a 2D array of shape (n_lagtimes, n_atoms)
+            for `non_linear=True`: a 2D array of shape (n_delta_t_values, n_atoms)
```

Comment:
The formatting does not look good when rendered, see [MDAnalysis.analysis.msd.EinsteinMSD.results.msds_by_particle](https://mdanalysis--5066.org.readthedocs.build/en/5066/documentation_pages/analysis/msd.html#MDAnalysis.analysis.msd.EinsteinMSD.results.msds_by_particle) in the docs. Please change the formatting so that it makes sense to a reader.

Always look at the rendered docs. You can read more about reST formatting with sphinx in the [sphinx reST primer](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html) and accompanying docs.

<img width="648" height="280" alt="image" src="https://github.com/user-attachments/assets/dcd4d528-9f36-4550-958e-09777bab372a" />